core/services/pipeline/orm.go

```diff
@@ -725,15 +720,15 @@ func (o *orm) prune(tx sqlutil.DataSource, jobID int32) {
 	if val%every == 0 {
 		ok := o.IfStarted(func() {
 			o.wg.Add(1)
-			go func() {
+			go func(ctx context.Context) {
 				o.lggr.Debugw("Pruning runs", "jobID", jobID, "count", val, "every", every, "maxSuccessfulRuns", o.maxSuccessfulRuns)
 				defer o.wg.Done()
-				ctx, cancel := context.WithTimeout(sqlutil.WithoutDefaultTimeout(o.ctx), time.Minute)
+				ctx, cancel := o.stopCh.CtxCancel(context.WithTimeout(sqlutil.WithoutDefaultTimeout(ctx), time.Minute))
```

Comment:
`time.Minute` seems like something that should be configured :), out of scope

Do we want our sqlutil timeout and context timeouts to be different keys? Is that why we need both invocations here to end up with the right timeout that consumers will respect?

Reply:
`WithoutDefaultTimeout` just tells the downstream WrappedDB Monitor not to apply the default timeout. Setting our own timeout is optional, and using one minute preserves the old behavior from `LongQueryTimeout` or whatever that was called. Normal timeouts are not communicated via key/vals, so these are not obviously combinable AFAICT. Although we could provider a helper for this kind of case where we do both :shrug: